src/RulesetEngine.php

```diff
@@ -59,4 +61,19 @@ public function process(
 
         return new Context($rules);
     }
+
+    private function isInternalClass(string $class): bool
+    {
+        if (is_a($class, \Throwable::class, true)) {
```

Comment:
I think this doesn't work with custom exceptions which extends `\Exception` or implements `\Throwable`. Would be `interface_exists` an alternative?

Reply:
You are right, it's fixed now.